src/EventListener/GiftCardCreationSubscriber.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Setono\SyliusGiftCardPlugin\EventListener;
+
+use Setono\SyliusGiftCardPlugin\Model\GiftCardInterface;
+use Setono\SyliusGiftCardPlugin\Provider\GiftCardChannelConfigurationProviderInterface;
+use Sylius\Bundle\ResourceBundle\Event\ResourceControllerEvent;
+use Sylius\Component\Resource\Exception\UnexpectedTypeException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+final class GiftCardCreationSubscriber implements EventSubscriberInterface
+{
+    /** @var GiftCardChannelConfigurationProviderInterface */
+    private $configurationProvider;
+
+    public function __construct(GiftCardChannelConfigurationProviderInterface $configurationProvider)
+    {
+        $this->configurationProvider = $configurationProvider;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            'setono_sylius_gift_card.gift_card.pre_create' => 'onGiftCardPreCreate',
+        ];
+    }
+
+    public function onGiftCardPreCreate(ResourceControllerEvent $event): void
+    {
+        $giftCard = $event->getSubject();
+
+        if (!$giftCard instanceof GiftCardInterface) {
+            throw new UnexpectedTypeException(
+                $giftCard,
+                GiftCardInterface::class
+            );
+        }
+
+        $configuration = $this->configurationProvider->getConfigurationForGiftCard($giftCard);
+        $giftCard->setConfiguration($configuration);
```

Comment:
Do we need to associate a gift card with a configuration? It only applies when we want to download a PDF, right?

Reply:
Yes, but for immutable concept, I believe we should attach them.
A customer should expect to get his gift card the same way it was even if we change the image for new cards. WDYT ?

Reply:
I agree, but it's still not immutable that way. Because the configuration is a reference I guess? :)

Reply:
Good point.
Then ideed the relation here is just a 'shortcut' and is not really needed. We can only rely on the provider to have the information